test/testsuite.h

```diff
@@ -53,6 +55,28 @@ class TestFixture : public ErrorLogger {
 
     bool assert_(const char * const filename, const unsigned int linenr, const bool condition) const;
 
+    template <typename T, typename U>
+    bool assertEquals(const char* const filename, const unsigned int linenr, const T& expected, const U& actual, const std::string& msg = emptyString) const {
+        if (expected != actual) {
+            ++fails_counter;
+
+            std::ostringstream expectedStr;
+            expectedStr << expected;
+            std::ostringstream actualStr;
+            actualStr << actual;
+
+            errmsg << getLocationStr(filename, linenr) << ": Assertion failed. " << std::endl
```

Comment:
I don't like that all this code is in the header. If we break out some stuff into a function and put that in the cpp file instead that is preferable.